dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/maintenance/jdbc/JdbcMaintenanceStore.java

```diff
@@ -137,6 +138,33 @@ public int deleteSoftDeletedProgramStageInstances()
         return result;
     }
 
+    @Override
+    public int deleteSoftDeletedRelationships()
+    {
+
+        List<String> deletedRelationships = getDeletionEntities(
+            "(select uid from relationship where deleted is true)" );
```

Comment:
When is the deletion going to run? Asking since I don't know how many rows this might return, from looking at this it could be many 🤷🏻 Do we have a mechanism for "pacing" such an operation like a DB rate limiter 😋 Or is this something only an admin will do, so they will take care and do this when appropriate.

Reply:
Only admins will do this. But the number of records is hard to predict. I would expect people only to do this in special cases. We dont know the number of rows though, so its not a bad idea to think about how to pace this. But again, as my previous comment, this would be relevant for all of these "hard delete soft deleted values" actions.